src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -234,25 +235,27 @@ internal void ApplyHttpsDefaults(HttpsConnectionAdapterOptions httpsOptions)
     /// </summary>
     public void Load()
     {
-        if (_loaded)
+        if (_reloadToken is null || _reloadToken.HasChanged)
         {
-            // The loader has already been run.
-            return;
+            // Will update _reloadToken
+            _ = Reload();
         }
-        _loaded = true;
-
-        Reload();
 
         foreach (var action in EndpointsToAdd)
         {
             action();
```

Comment:
Can we add a test for what happens a user-defined `Action<ListenOptions>` throws out of here?

Previously, if someone manually called `Load()` and swallowed the exception, we'd bind to everything defined before the throwing-action. I don't think it's important to preserve that behavior. Failing to bind is fine, but it'd be good to have a test for it.

It could lead to some really weird conflicts if there are any actions that throw non-deterministically, so it's probably best to avoid ever rerunning an action.

Reply:
I've added such a test and added a flag to prevent re-runs.